usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,686 @@
+        parent_uei,
+        state,
+        zip4,
+        zip5,
+        alternate_names,
+        source,
+        update_date,
+        row_num_union
+    )
+    SELECT
+        recipient_hash,
+        duns_recipient_hash,
+        legal_business_name,
+        duns,
+        uei,
+        address_line_1,
+        address_line_2,
+        business_types_codes,
+        city,
+        congressional_district,
+        country_code,
+        parent_duns,
+        parent_legal_business_name,
+        parent_uei,
+        state,
+        zip4,
+        zip5,
+        alternate_names,
+        source,
+        update_date,
+        row_num_union
+    FROM union_all_priority
+    """,
+    # -----
+    # Update the temporary_restock_recipient_lookup table to include any alternate names
+    # -----
+    r"""
+    MERGE INTO temp.temporary_restock_recipient_lookup temp_rl
+    USING (
+        WITH alt_names AS (
+            SELECT
+                recipient_hash,
+                COLLECT_SET(awardee_or_recipient_legal) AS all_names
+            FROM temporary_transaction_recipients_view
+            WHERE COALESCE(awardee_or_recipient_legal, '') != ''
+            GROUP BY recipient_hash
+        ),
+        alt_parent_names AS (
+            SELECT
+                parent_recipient_hash AS recipient_hash,
+                COLLECT_SET(ultimate_parent_legal_enti) AS all_names
+            FROM temporary_transaction_recipients_view
+            WHERE COALESCE(ultimate_parent_legal_enti, '') != ''
+            GROUP BY parent_recipient_hash
+        )
+        SELECT
+            COALESCE(an.recipient_hash, apn.recipient_hash) AS recipient_hash,
+            COALESCE(
+                ARRAY_SORT(ARRAY_UNION(COALESCE(an.all_names, ARRAY()), COALESCE(apn.all_names, ARRAY()))),
+                ARRAY()
+             )AS all_names
+        FROM alt_names AS an
+        FULL OUTER JOIN alt_parent_names AS apn ON an.recipient_hash = apn.recipient_hash
+    ) AS alt_names
+    ON temp_rl.recipient_hash = alt_names.recipient_hash AND row_num_union = 1
+    WHEN MATCHED
+    AND temp_rl.alternate_names IS DISTINCT FROM ARRAY_REMOVE(alt_names.all_names, COALESCE(temp_rl.legal_business_name, ''))
+    THEN UPDATE SET temp_rl.alternate_names = COALESCE(ARRAY_REMOVE(alt_names.all_names, COALESCE(temp_rl.legal_business_name, '')), ARRAY())
+    """,
+    # -----
+    # Alternate names defaults to an empty list
+    # -----
+    """
+    UPDATE temp.temporary_restock_recipient_lookup
+    SET alternate_names = ARRAY()
+    WHERE alternate_names IS NULL
+    """,
+    # -----
+    # Delete any instances of a NULL recipient_hash and handle cleanup following alternate names generation
+    # -----
+    r"""
+    DELETE FROM temp.temporary_restock_recipient_lookup
+    WHERE recipient_hash IS NULL OR row_num_union != 1
+    """,
+    # -----
+    # Merge the temporary_restock_recipient_lookup table into recipient_lookup
+    # -----
```

Comment:
Similar comment to one I made on Emily's PR:
>What's the purpose of populating a temporary table, temporary_restock_recipient_profile, then performing a series of merges to make recipient_profile match?
I believe this was done in Postgres to prevent a live table from being updated by a series of steps (1-9) that could fail, but because this DBR ETL is updating a table that will later be nuke/paved, would it make sense to clear recipient_profile at the start of the script and update directly? 

Any reason not to be populating Recipient Lookup directly instead of the temporary table?

Reply:
Following up on discussions from Slack.

Current thinking is that utilizing a separate temp table allows for slightly different columns between recipient_lookup and the temp_recipient_lookup without persisting those columns into DBR. The approach was changed however to use `INSERT OVERWRITE` instead of `MERGE INTO` in hopes that blowing away the old table and then inserting will be more performant than trying to go row by row.

Reply:
Ditto. 

basically could the temp table be the final table? It would exist in a half-baked state during the pipeline, but perhaps we have error handling that rolls it back to the previous version if a problem occurred

Reply:
Addressed this following example that Keith had shared to utilize a series of VIEWs.